test/control-loop.cc

```diff
@@ -0,0 +1,50 @@
+#include "eventuals/control-loop.h"
+
+#include "eventuals/collect.h"
+#include "eventuals/loop.h"
+#include "eventuals/map.h"
+#include "eventuals/pipe.h"
+#include "eventuals/repeat.h"
+#include "eventuals/until.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using eventuals::Collect;
+using eventuals::ControlLoop;
+using eventuals::Loop;
+using eventuals::Map;
+using eventuals::Pipe;
+using eventuals::Repeat;
+using eventuals::Until;
+
+using testing::ElementsAre;
+
+TEST(ControlLoop, SimplePipeHandling) {
```

Comment:
Please add a test that interrupts the control loop and shows that you can still wait for it and it will complete! You can use an `Eventual`, see tests/eventual.cc for some examples.